NIX.md

```diff
@@ -0,0 +1,181 @@
+
+## BDK's Nix-based tests and CI
+
+**TLDR: To replicate everything that our CI checks just do `nix flake check`**.
+
+BDK's tests and checks are run using Nix.
+If you want to run the tests locally, you can do so with:
+
+```text
+nix flake -L check --keep-failed
+```
+
+The `-L` flag prints full build logs on standard error.
+This is good for debugging.
+The `--keep-failed` flag keep temporary directories of failed builds.
+This can be useful for caching intermediate results while fixing failed checks.
+
+Under the hood `nix flake check` does the following:
+
+- Checks for typos, GPG-signed and conventional commits, `rustfmt`, `nixpkgs-fmt` (`.nix` files).
+- Runs `cargo clippy` in all workspace.
+- Runs `cargo fmt` in all workspace.
+- Checks dependencies for security advisory using [`rustsec/advisory-db`](https://github.com/rustsec/advisory-db).
+- `cargo build` and `cargo test` in whole workspace using the latest Rust version.
+- `cargo build` and `cargo test` in whole workspace using the MSRV Rust version.
+- `cargo build` and `cargo check` WASM targets using the latest Rust version
+   on `bdk` and `bdk_esplora` crates.
+
+The checks are by default cached by Nix and subsequent runs use the cached results,
+respecting the Nix derivations' dependency graph.
+
+All the necessary dependencies, e.g. `bitcoind` and `esplora`,
+along with all necessary environment variables,
+are installed and available in the environment that Nix's checks runs in.
+
+Individual checks from the above list can be listed with `nix flake show`
+and will be displayed under the `checks` tree.
+You can run an individual check with:
+
+```text
+nix build -L .#checks.SYSTEM.CHECK --keep-failed
+```
+
+Where `SYSTEM` is the system you are running on,
+and `CHECK` is the name of the check you want to run.
+For example, in a MacOS Apple Silicon machine (`aarch64-darwin`)
+ and for the `clippy` check you can run:
+
+```text
+nix build -L .#checks.aarch64-darwin.clippy --keep-failed
+```
+
+We also group all the CI checks inside the `legacyPackages`.
+These can be viewed in the `flake.nix` file under the `legacyPackages.ci` set.
+You can run any of these with:
+
+```text
+nix build -L .#ci.CHECK --keep-failed
+```
+
+Where `CHECK` is the name of the check you want to run.
+For example to run the `pre-commit-check`:
+
+```text
+nix build -L .#ci.pre-commit-check --keep-failed
+```
+
+You can check all the tests that our CI performs by inspecting both
+the `flake.nix` and `.github/workflows/cont_integration.yml` files.
+
+## Developer best-practices with Nix
+
+We provide several development shells,
+also called `devShell`s,
+that assist developers with the necessary dependency and environment
+to develop and test BDK.
+
+To access these shells, you can run:
+
+```text
+nix develop .#SHELL
+```
+
+Where `SHELL` is the name of the `devShell` you want to use.
+We provide the following shells:
+
+- default: latest Rust, you can omit the `#SHELL` and just run `nix develop .`.
```

Comment:
**suggestion:** `.` is not required, AFAIR.

Reply:
Yes, I agree: less is more.
But I am still preferring to show like this because teaches the `.#SHELL` logic better.
The idea is not to teach the user that they can suppress 2 characters, ` .`, when instantiating the default shell,
but to teach them the logic behind `.#SHELL1`, `.#SHELL2` and so on.